File: 703474-smallest-greater-elements-in-whole-array-GFG/703474-smallest-greater-elements-in-whole-array.java
// User function Template for Java

// User function Template for Java

// User function Template for Java

class Complete {
	
	// Function for finding maximum and value pair
	public static int[] greaterElement(int arr[], int n) {
		// Complete the function
		TreeSet<Integer> set = new TreeSet<>();
		
		for (int num : arr) {
			set.add(num);
		}
		
		int[] ans = new int[n];
		
		for (int i = 0; i < n; i++) {
			Integer next = set.higher(arr[i]);
			ans[i] = (next == null) ? -10000000 : next;
		}
		
		return ans;
	}
}


// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna

File: 703474-smallest-greater-elements-in-whole-array-GFG/703474-smallest-greater-elements-in-whole-array.cpp
int* greaterElement(int arr[], int n) {
	// Complete the function
	set<int> st;
	
	for (int i = 0; i < n; i++) {
		st.insert(arr[i]);
	}
	
	int* ans = new int[n];
	
	for (int i = 0; i < n; i++) {
		auto it = st.upper_bound(arr[i]);
		
		if (it == st.end())
			ans[i] = -10000000;
		else
			ans[i] = *it;
	}
	
	return ans;
}


// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna